core/templates/dev/head/pages/exploration-editor-page/services/exploration-rights.service.spec.ts

```diff
@@ -27,73 +27,275 @@ import { UpgradedServices } from 'services/UpgradedServices';
+      editor_names: [],
+      voice_artist_names: [],
+      viewer_names: [],
+      status: 'private',
+      cloned_from: 'e1234',
+      community_owned: true,
+      viewable_if_private: true
+    }
+  };
+  var clearWarningsSpy = null;
+
   beforeEach(angular.mock.module('oppia'));
   beforeEach(angular.mock.module('oppia', function($provide) {
     $provide.value(
       'ExplorationDraftObjectFactory', new ExplorationDraftObjectFactory());
+    $provide.value(
+      'ExplorationDataService', {
+        explorationId: '12345',
+        data: {
+          version: 1
+        }
+      });
   }));
   beforeEach(angular.mock.module('oppia', function($provide) {
     var ugs = new UpgradedServices();
     for (let [key, value] of Object.entries(ugs.getUpgradedServices())) {
       $provide.value(key, value);
     }
   }));
+  beforeEach(angular.mock.inject(function($injector, $q) {
+    ers = $injector.get('ExplorationRightsService');
+    als = $injector.get('AlertsService');
+    $httpBackend = $injector.get('$httpBackend');
+    CsrfService = $injector.get('CsrfTokenService');
 
-  describe('exploration rights service', function() {
-    var ers = null;
-
-    beforeEach(angular.mock.inject(function($injector) {
-      ers = $injector.get('ExplorationRightsService');
-    }));
-
-    it('correctly initializes the service', function() {
-      expect(ers.ownerNames).toBeUndefined();
-      expect(ers.editorNames).toBeUndefined();
-      expect(ers.voiceArtistNames).toBeUndefined();
-      expect(ers.viewerNames).toBeUndefined();
-      expect(ers._status).toBeUndefined();
-      expect(ers._clonedFrom).toBeUndefined();
-      expect(ers._isCommunityOwned).toBeUndefined();
-      expect(ers._viewableIfPrivate).toBeUndefined();
-
-      ers.init(['abc'], [], [], [], 'private', 'e1234', true, true);
-
-      expect(ers.ownerNames).toEqual(['abc']);
-      expect(ers.editorNames).toEqual([]);
-      expect(ers.voiceArtistNames).toEqual([]);
-      expect(ers.viewerNames).toEqual([]);
-      expect(ers._status).toEqual('private');
-      expect(ers._clonedFrom).toEqual('e1234');
-      expect(ers._isCommunityOwned).toBe(true);
-      expect(ers._viewableIfPrivate).toBe(true);
+    spyOn(CsrfService, 'getTokenAsync').and.callFake(function() {
+      var deferred = $q.defer();
+      deferred.resolve('sample-csrf-token');
+      return deferred.promise;
     });
 
-    it('reports the correct cloning status', function() {
-      ers.init(['abc'], [], [], [], 'public', '1234', true);
-      expect(ers.isCloned()).toBe(true);
-      expect(ers.clonedFrom()).toEqual('1234');
+    clearWarningsSpy = spyOn(als, 'clearWarnings').and.callThrough();
+  }));
+
+  it('should correctly initializes the service', function() {
+    expect(ers.ownerNames).toBeUndefined();
+    expect(ers.editorNames).toBeUndefined();
+    expect(ers.voiceArtistNames).toBeUndefined();
+    expect(ers.viewerNames).toBeUndefined();
+    expect(ers.isPrivate()).toBe(false);
```

Comment:
I like that you have removed usage of private functions/variables :)

Reply:
Thanks!